src/session.ts

```diff
@@ -480,7 +481,11 @@ export class Session extends common.GrpcServiceObject {
     options?: TimestampBounds,
     queryOptions?: google.spanner.v1.ExecuteSqlRequest.IQueryOptions
   ) {
-    return new Snapshot(this, options, queryOptions);
+    const span = getActiveOrNoopSpan();
+    span.addEvent('Creating Snapshot');
+    const snapshot = new Snapshot(this, options, queryOptions);
+    span.addEvent('Snapshot creation done', {id: snapshot?.id?.toString()});
```

Comment:
Again, the transaction is not getting created here. Add this line, only for the places where we are calling "transaction.begin"

Reply:
like this place 
https://github.com/googleapis/nodejs-spanner/blob/main/src/database.ts#L2033

and this https://github.com/googleapis/nodejs-spanner/blob/main/src/database.ts#L2773

I would recommend to add below event at https://github.com/googleapis/nodejs-spanner/blob/main/src/transaction.ts#L1419 . This method gets called from all the places whenever transaction gets created and assigns the id. This would mean you can remove the `BatchTransaction creation done` event as well.

`span.addEvent('Transaction creation done', {id: id?.toString()});`